src/components/OpenWith/OpenWith.js

```diff
@@ -192,9 +200,64 @@ class OpenWith extends PureComponent<Props, State> {
      * @private
      * @return {void}
      */
-    onIntegrationClick(): void {
-        /* no-op */
-    }
+    onIntegrationClick = (appIntegrationId: string): void => {
+        const { fileId }: Props = this.props;
+        this.api
+            .getAppIntegrationsAPI(false)
+            .execute(
+                appIntegrationId,
+                fileId,
+                this.executeIntegrationSuccessHandler,
+                this.executeIntegrationErrorHandler,
+            );
+    };
+
+    /**
+     * Opens the integration in a new tab based on the API data
+     *
+     * @private
+     * @param {ExecuteAPI} executeData - API response on how to open an executed integration
+     * @return {void}
+     */
+    executeIntegrationSuccessHandler = (executeData: ExecuteAPI) => {
+        switch (executeData.method.toLowerCase()) {
+            case HTTP_POST:
+                this.setState({ executePostData: executeData }, () => {
+                    const form: ?HTMLFormElement = document.querySelector(
+                        `#bcow-execute-form-${this.id}`,
+                    );
+
+                    if (!form) {
+                        this.executeIntegrationErrorHandler(
+                            Error('execution form unavailable'),
```

Comment:
Message as a constant

Reply:
Will !form ever be true?

Reply:
that depends on if setstate can fail to invoke the callback, not sure if that's possible